lib/compat/wordpress-6.0/register-webfonts-from-theme-json.php

```diff
@@ -122,28 +123,28 @@ function gutenberg_add_registered_webfonts_to_theme_json( $data ) {
 		$data['settings']['typography']['fontFamilies'] = array();
 	}
 
-	// Add missing fonts.
-	foreach ( $to_add as $family ) {
-		$font_face = array();
-		foreach ( $font_families_registered as $font_family ) {
-			if ( $family !== $font_family['font-family'] ) {
-				continue;
-			}
-			$camel_cased = array();
-			foreach ( $font_family as $key => $value ) {
+	foreach ( $to_add as $slug ) {
+		$font_faces_for_family = $font_families_registered[ $slug ];
+		$family_name           = $font_faces_for_family[0]['font-family'];
+		$font_faces            = array();
+
+		foreach ( $font_faces_for_family as $font_face ) {
+			$camel_cased = array( 'origin' => 'gutenberg_wp_webfonts_api' );
+			foreach ( $font_face as $key => $value ) {
 				$camel_cased[ lcfirst( str_replace( '-', '', ucwords( $key, '-' ) ) ) ] = $value;
 			}
-			$font_face[] = $camel_cased;
+			$font_faces[] = $camel_cased;
 		}
+
 		$data['settings']['typography']['fontFamilies'][] = array(
-			'fontFamily' => false !== strpos( $family, ' ' ) ? "'{$family}'" : $family,
-			'name'       => $family,
-			'slug'       => sanitize_title( $family ),
-			'fontFace'   => $font_face,
+			'fontFamily' => false !== strpos( $family_name, ' ' ) ? "'{$family_name}'" : $family_name,
+			'name'       => $family_name,
+			'slug'       => $slug,
+			'fontFace'   => $font_faces,
 		);
 	}
 
 	return $data;
 }
 
-add_action( 'wp_loaded', 'gutenberg_register_webfonts_from_theme_json' );
+add_action( 'init', 'gutenberg_register_webfonts_from_theme_json' );
```

Comment:
@aristath any specific reason to hook into `wp_loaded` instead of `init`?

`init` runs earlier but after the theme is loaded. Moving to `init` makes enqueueing fonts in `wp_loaded` (and `init`) possible. Any concerns?

Reply:
I don't recall exactly why we used `wp_loaded` here... 🤔 
If we change this, we need to check the following:
* Registering fronts both via `theme.json` and PHP (using the `wp_register_webfonts` function)
* Registered webfonts (via theme.json and PHP) appear in the editor's font-family selector
* CSS for registered webfonts (via theme.json and PHP) is present in the editor
* CSS for registered webfonts (via theme.json and PHP) is present ojn the frontend

If all of the above work as expected, then there's no issue switching the hook to `init` 👍

Reply:
Great 👍

Nothing changed, so we're good to go.